File: 557.Reverse-Words-In-A-String/reverse-word-in-a-string.js
// Given a string s, reverse the order of characters in each word within a sentence while still preserving whitespace and initial word order. 

// Example 1:
// Input: s = "Let's take LeetCode contest"
// Output: "s'teL ekat edoCteeL tsetnoc"

// Example 2:
// Input: s = "God Ding"
// Output: "doG gniD"

const reverseWords = function(string) {
    let arrayOfStrings = string.split(" ");
    const reversedArray = arrayOfStrings.map(word => word.split("").reverse().join(""));
    let reversedString = reversedArray.join(" ")
    return reversedString
};

File: 557.Reverse-Words-In-A-String/reverse-word-in-a-string.py
class Solution:
    def reverseWords(self, string: str) -> str:
        # Firstly, I split the string to generate an array of the words
        arrayOfStrings = string.split(" ")
        
        # I reverse each of the words that are in the array 
        reversedArray = [word[::-1] for word in arrayOfStrings]

        # I join every element of the array
        reversedString = " ".join(reversedArray)
        
        return reversedString
    